parser/object_parse/die_parse/mod.rs

```diff
@@ -56,22 +56,26 @@ where
             comp_unit,
         } = extra;
 
-        let item = match item_kind(self, debug_types, comp_unit)? {
+        let id = ir::Id::entry(unit_id, entry_id);
+
+        let item: ir::Item = match item_kind(self, debug_types, comp_unit)? {
             Some(kind @ ir::ItemKind::Subroutine(_)) => {
                 let name = item_name(self, debug_str)?
                     .unwrap_or(format!("Subroutine[{}][{}]", unit_id, entry_id));
-                let id = ir::Id::entry(unit_id, entry_id);
-                DieLocationAttributes::try_from(self)?
-                    .entity_size(addr_size, dwarf_version, rnglists)?
-                    .map(|size| ir::Item::new(id, name, size as u32, kind))
+                if let Some(size) = DieLocationAttributes::try_from(self)?.entity_size(
+                    addr_size,
+                    dwarf_version,
+                    rnglists,
+                )? {
+                    ir::Item::new(id, name, size as u32, kind)
+                } else {
+                    return Ok(());
+                }
```

Comment:
I added early returns here, I think it helped clean things up a *little*. It's worth noting that the `entity_size` method here returns an Option because (if I understand correctly) not all `DW_AT_subprogram` DIEs are describing a definition. These DIEs don't include the same location attributes, in which case this method returns None.